packages/slate-plugins/src/elements/mention/useMention.tsx

```diff
@@ -1,121 +1,81 @@
+import { useCallback, useState } from 'react';
+import { isPointAtWordEnd, isWordAfterTrigger } from 'common/queries';
+import { getNextIndex } from 'elements/mention/utils/getNextIndex';
+import { getPreviousIndex } from 'elements/mention/utils/getPreviousIndex';
 import { Editor, Range, Transforms } from 'slate';
-import { MentionSelect } from './components/MentionSelect';
 import { insertMention } from './transforms';
-import { MentionableItem, MentionOptions } from './types';
-
-const AFTER_MATCH_REGEX = /^(\s|$)/;
-
-const getMentionSelect = (
-  target: Range | null,
-  index: number,
-  mentionables: MentionableItem[]
-) => {
-  return () => {
-    if (target && mentionables.length > 0) {
-      return (
-        <MentionSelect
-          target={target}
-          index={index}
-          mentionables={mentionables}
-        />
-      );
-    }
-    return null;
-  };
-};
+import { MentionableItem, UseMentionOptions } from './types';
 
 export const useMention = (
   mentionables: MentionableItem[] = [],
-  options: Partial<MentionOptions>
+  { maxSuggestions = 10, trigger = '@' }: UseMentionOptions = {}
 ) => {
-  const { maxSuggestions = 10, trigger = '@', prefix = trigger } = options;
-  const [target, setTarget] = useState<Range | null>(null);
-  const [index, setIndex] = useState(0);
+  const [targetRange, setTargetRange] = useState<Range | null>(null);
+  const [valueIndex, setValueIndex] = useState(0);
   const [search, setSearch] = useState('');
-  const matchingMentionables = mentionables
+  const values = mentionables
     .filter((c) => c.value.toLowerCase().includes(search.toLowerCase()))
     .slice(0, maxSuggestions);
 
-  const MentionSelectComponent = getMentionSelect(
-    target,
-    index,
-    matchingMentionables
-  );
-
   const onKeyDownMention = useCallback(
     (e: any, editor: Editor) => {
-      if (target) {
-        switch (e.key) {
-          case 'ArrowDown': {
-            e.preventDefault();
-            const prevIndex =
-              index >= matchingMentionables.length - 1 ? 0 : index + 1;
-            setIndex(prevIndex);
-            break;
-          }
-          case 'ArrowUp': {
-            e.preventDefault();
-            const nextIndex =
-              index <= 0 ? matchingMentionables.length - 1 : index - 1;
-            setIndex(nextIndex);
-            break;
-          }
-          case 'Tab':
-          case 'Enter':
-            e.preventDefault();
-            Transforms.select(editor, target);
-            insertMention(editor, matchingMentionables[index], prefix);
-            setTarget(null);
-            break;
-          case 'Escape':
-            e.preventDefault();
-            setTarget(null);
-            break;
-          default:
-            break;
+      if (targetRange) {
+        if (e.key === 'ArrowDown') {
+          e.preventDefault();
+          setValueIndex(getNextIndex(valueIndex, values.length - 1));
+        }
+        if (e.key === 'ArrowUp') {
```

Comment:
I think it would be safer to use if-else here, also this way we evaluate 3 more conditions on each run